tests/validation/cattlevalidationtest/core/test_services.py

```diff
@@ -908,3 +936,24 @@ def wait_until_instances_get_stopped(super_client, service, timeout=60):
         if time.time() - start > timeout:
             raise Exception(
                 'Timed out waiting for instances to get to stopped state')
+
+
+def wait_for_scale_to_adjust(super_client, service):
+    instance_maps = super_client.list_serviceExposeMap(serviceId=service.id,
```

Comment:
You might want to put service = client.wait_success(service), so the check for instance maps will be performed only when the service is settled.